### PR TITLE
[pytorch][mobile] change mobile build scripts to build PyTorch by default

### PR DESCRIPTION
**Pull request by ljk53**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34203 [pytorch][mobile] change mobile build scripts to build PyTorch by default**

Summary:
Currently cmake and mobile build scripts still build libcaffe2 by
default. To build pytorch mobile users have to set environment variable
BUILD_PYTORCH_MOBILE=1 or set cmake option BUILD_CAFFE2_MOBILE=OFF.

PyTorch mobile has been released for a while. It's about time to change
CMake and build scripts to build libtorch by default.

Changed caffe2 CI job to build libcaffe2 by setting BUILD_CAFFE2_MOBILE=1
environment variable. Only found android CI for libcaffe2 - do we ever
have iOS CI for libcaffe2?

Differential Revision: [D20267274](https://our.internmc.facebook.com/intern/diff/D20267274)

### CODE REVIEW

File: .jenkins/caffe2/build.sh [hunk 1]
```diff
@@ -104,7 +104,7 @@ if [[ "${BUILD_ENVIRONMENT}" == *-android* ]]; then
   build_args+=("BUILD_TEST=ON")
   build_args+=("USE_OBSERVERS=ON")
   build_args+=("USE_ZSTD=ON")
-  "${ROOT_DIR}/scripts/build_android.sh" $(build_to_cmake ${build_args[@]}) "$@"
+  BUILD_CAFFE2_MOBILE=1 "${ROOT_DIR}/scripts/build_android.sh" $(build_to_cmake ${build_args[@]}) "$@"
```
Comment (ljk53) [inline .jenkins/caffe2/build.sh]: Do we still run this CI job? Could find it in .circleci/config.yml

**Comment by ezyang** [comment]
Pretty sure it's Android only, but maybe @pietern would know better

File: .jenkins/caffe2/build.sh (hunk 1 above)
Comment (ezyang) [reply .jenkins/caffe2/build.sh]: It's still used, see `caffe2_linux_build`

**Review by ezyang — APPROVED** [review]

File: .jenkins/caffe2/build.sh (hunk 1 above)
Comment (ljk53) [reply .jenkins/caffe2/build.sh]: > caffe2_linux_build

right, but caffe2_linux_build doesn't seem to have any job with `"${BUILD_ENVIRONMENT}" == *-android*` (I meant to ask about the caffe2 android CI)

**Pull request closed without merging**